src/schoology-extractor/schoology_extractor/__main__.py

```diff
@@ -33,23 +35,44 @@
 page_size = arguments.page_size
 input_directory = arguments.input_directory
 
-# Configure logging
-logFormatter = "%(asctime)s - %(levelname)s - %(name)s - %(message)s"
+db_engine: sqlalchemy.engine.base.Engine
+extractorFacade: SchoologyExtractFacade
 
-logging.basicConfig(
-    handlers=[
-        logging.StreamHandler(sys.stdout),
-    ],
-    format=logFormatter,
-    level=log_level,
-)
+logger: logging.Logger
+error_tracker: ErrorHandler
 
-logger = logging.getLogger(__name__)
 
-request_client = RequestClient(schoology_key, schoology_secret)
-db_engine = get_sync_db_engine()
+def _configure_logging():
```

Comment:
I didn't like the logging and initialization stuff above that was just hanging out in global code, before the call to `main()`. So I wrapped in a couple of functions that are called _from_ `main`.

Reply:
I could go either way on this.  Logging seems kind of special in that logging setup is typically the very first thing you want to do in any app, and loggers are usually accessed globally anyway rather than passed in as a dependency.  Not a strong preference though.